airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -62,8 +62,9 @@ def __init__(
                 AirflowProviderDeprecationWarning,
                 stacklevel=3,  # 2 is in the operator's init, 3 is in the user code creating the operator
             )
-        self.region_name = region_name
-        self.hook = RdsHook(aws_conn_id=aws_conn_id, region_name=region_name, **(hook_params or {}))
+        hook_params = hook_params or {}
+        self.region_name = region_name or hook_params.pop("region_name", None)
+        self.hook = RdsHook(aws_conn_id=aws_conn_id, region_name=region_name, **(hook_params))
```

Comment:
Shouldn't this use `self.region_name` now that it is set?

Reply:
yeah right 🤦